moveit_planners/trajopt/test/trajectory_test.cpp

```diff
@@ -0,0 +1,220 @@
+/*********************************************************************
+ * Software License Agreement
+ *
+ *  Copyright (c) 2019, PickNik Consulting
+ *  All rights reserved.
+ *
+ *********************************************************************/
+
+/* Author: Omid Heidari
+   Desc:   Test the trajectory planned by trajopt
+ */
+
+// C++
+#include <iostream>
+#include <string>
+#include <cmath>
+
+// ROS
+#include <ros/ros.h>
+
+// Testing
+#include <gtest/gtest.h>
+
+#include <trajopt/common.hpp>
+
+#include <moveit/planning_interface/planning_interface.h>
+#include <moveit/robot_model_loader/robot_model_loader.h>
+#include <moveit/planning_scene/planning_scene.h>
+#include <moveit/robot_state/robot_state.h>
+#include <moveit/utils/robot_model_test_utils.h>
+
+#include <trajopt_interface/problem_description.h>
+#include <trajopt_interface/kinematic_terms.h>
+
+#include <moveit/kinematic_constraints/utils.h>
+#include <moveit/move_group_interface/move_group_interface.h>
+#include <moveit/robot_model_loader/robot_model_loader.h>
+#include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+#include <moveit_msgs/MotionPlanResponse.h>
+
+class TrajectoryTest : public ::testing::Test
+{
+public:
+  TrajectoryTest()
+  {
+  }
+
+protected:
+  void SetUp() override
+  {
+    node_handle_ = ros::NodeHandle("~");
+    robot_model_ = moveit::core::loadTestingRobotModel("panda");
+    bool robot_model_ok_ = static_cast<bool>(robot_model_);
+    if (!robot_model_ok_)
+      ROS_ERROR_STREAM_NAMED("trajectory_test", "robot model is not loaded correctly");
+  }
+
+protected:
+  robot_model::RobotModelPtr robot_model_;
+  std::vector<std::string> group_joint_names_;
+  const std::string PLANNING_GROUP_ = "panda_arm";
+  const double GOAL_TOLERANCE = 0.1;
+  ros::NodeHandle node_handle_;
+};  // class TrajectoryTest
+
+TEST_F(TrajectoryTest, concatVectorValidation)
+{
+  std::vector<double> vec_a = { 1, 2, 3, 4, 5 };
+  std::vector<double> vec_b = { 6, 7, 8, 9, 10 };
+  std::vector<double> vec_c = trajopt_interface::concatVector(vec_a, vec_b);
+  EXPECT_EQ(vec_c.size(), vec_a.size() + vec_b.size());
+
+  std::size_t length_ab = vec_a.size() + vec_b.size();
+  for (std::size_t index = 0; index < length_ab; ++index)
+  {
+    if (index < vec_a.size())
+    {
+      EXPECT_EQ(vec_c[index], vec_a[index]);
+    }
+    else
+    {
+      EXPECT_EQ(vec_c[index], vec_b[index - vec_a.size()]);
+    }
+  }
+}
+
+TEST_F(TrajectoryTest, goalTolerance)
+{
+  SetUp();
```

Comment:
SetUp should be called already before this method is entered by gtest, should remove.